stdlib/public/core/String.swift

```diff
@@ -445,6 +445,10 @@ extension String {
   public init<C: Collection, Encoding: Unicode.Encoding>(
     decoding codeUnits: C, as sourceEncoding: Encoding.Type
   ) where C.Iterator.Element == Encoding.CodeUnit {
+    guard codeUnits.count > 0 else {
```

Comment:
This check should really go SOMEWHERE, but maybe not here, we'll see

Reply:
Is there a reason to believe that there is an encoding-dependent mapping of an empty input that is not ""?
This should check against `codeUnits.isEmpty`, though.

Reply:
Nah correctness-wise it's fine, my concern was more about increasing the amount of code in the inlineable path. Will definitely fix to use isEmpty, thanks.